pylint/checkers/similar.py

```diff
@@ -360,27 +360,27 @@ def append_stream(
         else:
             readlines = stream.readlines  # type: ignore[assignment] # hint parameter is incorrectly typed as non-optional
         try:
-            active_lines: list[str] = []
-            if hasattr(self, "linter"):
-                # Remove those lines that should be ignored because of disables
-                for index, line in enumerate(readlines()):
-                    if self.linter._is_one_message_enabled("R0801", index + 1):  # type: ignore[attr-defined]
-                        active_lines.append(line)
-            else:
-                active_lines = readlines()
-
-            self.linesets.append(
-                LineSet(
-                    streamid,
-                    active_lines,
-                    self.ignore_comments,
-                    self.ignore_docstrings,
-                    self.ignore_imports,
-                    self.ignore_signatures,
-                )
-            )
+            lines = readlines()
         except UnicodeDecodeError:
             pass
+        ignored_linenos = set()
+        if hasattr(self, "linter"):
+            for i in range(1, len(lines) + 1):
```

Comment:
Another idea (let me know what you think) what about passing `_is_one_message_enabled` to `LineSet`? And then call it during the passing over all lines these. With `getattr(linter, "_is_one_message_enabled", None)` we should be able to call the line only if necessary. Feels a bit of a waste to iterate over all `lines` twice. No idea what the actual performance impact of this would be, but passing each line of the `home assistant` project twice (for example) seems like it could take some time.

Reply:
Yeah, I considered passing the whole `Linter`, but that's a nice compromise to just pass a function.

Reply:
That was initial idea as well, but the function might be a bit more elegant. Not sure what works best.